pydeseq2/ds.py

```diff
@@ -633,26 +653,38 @@ def _build_contrast(self, contrast: Optional[List[str]] = None) -> None:
                     f"The contrast variable ('{contrast[0]}') should be one "
                     f"of the design factors."
                 )
-            if contrast[1] not in self.dds.obs[contrast[0]].values:
-                raise KeyError(
-                    f"The tested level ('{contrast[1]}') should correspond to one of the"
-                    f" levels of '{contrast[0]}'"
-                )
-            if contrast[2] not in self.dds.obs[contrast[0]].values:
-                raise KeyError(
-                    f"The reference level ('{contrast[2]}') should correspond to one of"
-                    f" the levels of '{contrast[0]}'"
-                )
+            if not contrast[1] == contrast[2] == "":
```

Comment:
```suggestion
            if not (contrast[1] == contrast[2] == ""):
```
Just to be clear on the logical statement